bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/EMFFacetConstraints.java

```diff
@@ -0,0 +1,171 @@
+
+   public Integer getLength() { return length; }
+
+   public void setLength(final Integer length) { this.length = length; }
+
+   public Integer getMinLength() { return minLength; }
+
+   public void setMinLength(final Integer minLength) { this.minLength = minLength; }
+
+   public Integer getMaxLength() { return maxLength; }
+
+   public void setMaxLength(final Integer maxLength) { this.maxLength = maxLength; }
+
+   public String getMinExclusive() { return minExclusive; }
+
+   public void setMinExclusive(final String minExclusive) { this.minExclusive = minExclusive; }
+
+   public String getMaxExclusive() { return maxExclusive; }
+
+   public void setMaxExclusive(final String maxExclusive) { this.maxExclusive = maxExclusive; }
+
+   public String getMinInclusive() { return minInclusive; }
+
+   public void setMinInclusive(final String minInclusive) { this.minInclusive = minInclusive; }
+
+   public String getMaxInclusive() { return maxInclusive; }
+
+   public void setMaxInclusive(final String maxInclusive) { this.maxInclusive = maxInclusive; }
+
+   public EMFFacetConstraints(final Map<String, Object> facetMap) {
+      this.whiteSpace = (Integer) facetMap.getOrDefault(WHITESPACE, null);
+      this.enumeration = (List<String>) facetMap.getOrDefault(ENUMERATION, null);
+      this.pattern = (List<String>) facetMap.getOrDefault(PATTERN, null);
+      this.totalDigits = (Integer) facetMap.getOrDefault(TOTALDIGITS, null);
+      this.fractionDigits = (Integer) facetMap.getOrDefault(FRACTIONDIGITS, null);
+      this.length = (Integer) facetMap.getOrDefault(LENGTH, null);
+      this.minLength = (Integer) facetMap.getOrDefault(MINLENGTH, null);
+      this.maxLength = (Integer) facetMap.getOrDefault(MAXLENGTH, null);
+      this.minExclusive = (String) facetMap.getOrDefault(MINEXCLUSIVE, null);
+      this.maxExclusive = (String) facetMap.getOrDefault(MAXEXCLUSIVE, null);
+      this.minInclusive = (String) facetMap.getOrDefault(MININCLUSIVE, null);
+      this.maxInclusive = (String) facetMap.getOrDefault(MAXINCLUSIVE, null);
+   }
+
+   public EMFFacetConstraints() {
+
+   }
+
+}
+
+class WhiteSpaceFilter {
+   @Override
+   public boolean equals(final Object obj) {
+      if (obj instanceof Integer) {
+         return obj == Integer.valueOf(0);
+      }
+      return true;
+   }
+
+   @Override
+   public int hashCode() {
+      // TODO Auto-generated method stub
+      return super.hashCode();
+   }
+}
+
+class IntegerFilter {
+   @Override
+   public boolean equals(final Object obj) {
+      if (obj instanceof Integer) {
+         return obj == Integer.valueOf(-1);
+      }
+      return true;
+   }
+
+   @Override
+   public int hashCode() {
+      // TODO Auto-generated method stub
+      return super.hashCode();
+   }
+}
+
+class StringListFilter {
+   @Override
+   public boolean equals(final Object obj) {
+      if (obj instanceof List) {
+         return ((List) obj).isEmpty();
```

Comment:
```suggestion
         return ((List<?>) obj).isEmpty();
```
Please avoid warning here